src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/TrancheUnitTest.java

```diff
@@ -0,0 +1,98 @@
+package org.broadinstitute.hellbender.tools.walkers.vqsr;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.text.XReadLines;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class TrancheUnitTest extends BaseTest {
+    private final String testDir = CommandLineProgramTest.getTestDataDir() + "/TrancheManagerUnitTest/";
+    private final File QUAL_DATA = new File(testDir + "tranches.raw.dat");
+    private final double[] TRUTH_SENSITIVITY_CUTS = new double[]{99.9, 99.0, 97.0, 95.0};
+    private final File EXPECTED_TRANCHES_NEW = new File(testDir + "tranches.6.txt");
+    private final File EXPECTED_TRANCHES_OLD = new File(testDir + "tranches.4.txt");
+
+    private ArrayList<VariantDatum> readData() throws IOException{
+        ArrayList<VariantDatum> vd = new ArrayList<>();
+        try (XReadLines xrl = new XReadLines(QUAL_DATA, true)){
+            for ( String line : xrl ) {
+                String[] parts = line.split("\t");
+                // QUAL,TRANSITION,ID,LOD,FILTER
+                if ( ! parts[0].equals("QUAL") ) {
+                    VariantDatum datum = new VariantDatum();
+                    datum.lod = Double.valueOf(parts[3]);
+                    datum.isTransition = parts[1].equals("1");
+                    datum.isKnown = ! parts[2].equals(".");
+                    datum.isSNP = true;
+                    datum.atTruthSite = datum.isKnown;
+                    vd.add(datum);
+                }
+            }
+        }
+
+        return vd;
+    }
+
+    @Test(expectedExceptions = {UserException.MalformedFile.class})
+    public final void readBadFormat() throws IOException{
+        Tranche.readTranches(QUAL_DATA);
+    }
+
+    @Test
+    public final void readNewFormat() throws IOException{
+        read(EXPECTED_TRANCHES_NEW);
+    }
+
+    @Test(expectedExceptions = {UserException.MalformedFile.class})
+    public final void readOldFormat() throws IOException {
+        read(EXPECTED_TRANCHES_OLD);
+    }
+
+    public final List<Tranche> read(File f) throws IOException{
+        return Tranche.readTranches(f);
+    }
+
+    private static void assertTranchesAreTheSame(List<Tranche> newFormat, List<Tranche> oldFormat, boolean completeP, boolean includeName) {
```

Comment:
probably not - we never put them into collections etc. And The test needs a special trick to ignore the name so it'd need a different equals which is not worth it.